Tests/PenguinStructuresTests/ArrayStorageTests.swift

```diff
@@ -78,6 +78,80 @@ extension ArrayStorageImplementation where Element: Equatable {
       }
     }
   }
+
+  /// Tests `appending`, or if `typeErased == true`, `appendingValue(at:)`.
+  static func test_appending<Source: Collection>(
+    source: Source, typeErased: Bool = false
+  )
+    where Source.Element == Element
+  {
+    for n in 0..<(source.count + 2) {
+      let s = Self.create(minimumCapacity: n)
+      
+      func doAppending(_ e: Element, moveElements: Bool) -> Self {
+        let saveCapacity = s.capacity
+        let saveCount = s.count
+        let r = typeErased
+          ? withUnsafePointer(to: e) {
+            s.appendingValue(at: .init($0), moveElements: moveElements)
+          }
+          : s.appending(e, moveElements: moveElements)
+        if saveCount == saveCapacity {
+          XCTAssertGreaterThanOrEqual(r.capacity, s.capacity * 2)
+        }
+        else {
+          XCTAssertEqual(r.capacity, s.capacity)
+        }
+        return r
+      }
+      
+      for e in source.prefix(n) {
+        _ = s.append(e)
+      }
+      let saveCount = s.count
+      let s1 = doAppending(source.first!, moveElements: false)
+      XCTAssertEqual(s.count, saveCount)
+      
+      s.withUnsafeMutableBufferPointer { b in
+        s1.withUnsafeMutableBufferPointer { b1 in
+          XCTAssertEqual(b1.count, b.count + 1)
+          XCTAssert(b1.dropLast().elementsEqual(b))
+          XCTAssertEqual(b1.last, source.first)
+        }
+      }
+      
+      let s2 = doAppending(source.first!, moveElements: true)
+      
+      s1.withUnsafeMutableBufferPointer { b1 in
+        s2.withUnsafeMutableBufferPointer { b2 in
+          XCTAssert(b1.elementsEqual(b2))
+        }
+      }
+    }
+  }
+
+  static func test_replacementStorage<Source: Collection>(source: Source)
```

Comment:
Nit: starting with `test_` makes it seem to me that this is a test function that happens to be in the wrong place. Please consider some other prefix (e.g. verify).

Reply:
I don't understand the objection.  It *is* a test function; you do see the XCTAssert calls in there.  Don't know what you mean by "the wrong place".  Also, this is consistent with other names in the file.  Are you suggesting that they all be changed?

Reply:
@saeta bump